components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt

```diff
@@ -179,6 +185,16 @@ abstract class EngineSession(
      */
     abstract fun clearFindMatches()
 
+    /**
+     * Enables/disables fullscreen mode. It also invokes a callback when fullscreen mode is exited.
+     * For GeckoView, this shouldn't necessarily be used to
+     *
+     * @param enable Turns fullscreen mode on/off.
+     * @param exitCallback (Optional) Invoked when the engine exits fullscreen mode. This is mostly
+     * needed for only SystemWebView implementations which need to invoke WebView APIs on exiting.
+     */
+    abstract fun fullScreenMode(enable: Boolean, exitCallback: () -> Unit? = {})
```

Comment:
If we are able to hide the View inside the engine (see above) then we may be able to do the same for the callback.